cura/PrinterOutput/ExtruderOutputModel.py

```diff
@@ -68,3 +74,13 @@ def updateHotendID(self, id: str):
         if self._hotend_id != id:
             self._hotend_id = id
             self.hotendIDChanged.emit()
+
+    @pyqtProperty(QObject, notify = extruderConfigurationChanged)
+    def extruderConfiguration(self):
+        return self._extruder_configuration
+
+    def _updateExtruderConfiguration(self):
```

Comment:
Why is this updating done in a separate step? Instead of setting the position, while setting self._position, material while setting self._active_material and hotendID while setting _hotend_id? Then it is never out of sync and maybe you don't need the signal.

Reply:
I remove the listeners and set the values when the material and hotend are set. Done in dfb903fb8113b2694085d95f80d3982ab4707a16